src/ecHiTacticals.ml

```diff
@@ -117,7 +117,7 @@ and process1_seq (ttenv : ttenv) (ts : ptactic list) (tc : tcenv1) =
 and process1_nstrict (ttenv : ttenv) (t : ptactic_core) (tc : tcenv1) =
   if ttenv.tt_smtmode <> `Strict then
     tc_error !!tc "try! can only be used in strict proof mode";
```

Comment:
Can I check that this indeed will fail out if a script is checked in weak mode? (I guess this then sets `tt_smtmode` to `Admit`, but I can't trace the logic back with the time I have now.)

Reply:
This will fail in weak check mode. The `Strict` here is the SMT option, not the proof mode.